### PR TITLE
Update code to consider escape character for script generation for entity names and reference types

### PR DESCRIPTION
**Pull request by sahanHe**
## Purpose

Fixes:[4034](https://github.com/ballerina-platform/ballerina-standard-library/issues/4034)

## Checklist
- [x] Linked to an issue
- [ ] ~Updated the specification~
- [ ] ~Updated the changelog~
- [x] Added tests

### CODE REVIEW

File: persist-cli/src/main/java/io/ballerina/persist/utils/SqlScriptGenerationUtils.java [hunk 1]
```diff
@@ -37,6 +37,7 @@
 import java.util.Set;
 import java.util.stream.Collectors;
 
+
```
Comment (kalaiyarasiganeshalingam) [inline persist-cli/src/main/java/io/ballerina/persist/utils/SqlScriptGenerationUtils.java:40]:
```suggestion
```

File: persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxGenerator.java [hunk 2]
```diff
@@ -1048,17 +1053,20 @@ private static ModuleMemberDeclarationNode createUpdateRecord(Entity entity) {
...
         }
+        if (entityName.startsWith(SINGLE_QUOTE)) {
+            return SINGLE_QUOTE + outputString.toString();
+        }
         return outputString.toString();
     }
 
-    private static String stripEscapeCharacter(String fieldName) {
-        return fieldName.startsWith("'") ? fieldName.substring(1) : fieldName;
+    public static String stripEscapeCharacter(String fieldName) {
```
Comment (kalaiyarasiganeshalingam) [inline persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxGenerator.java:1069]: This method seems to be used only in this file. Do we have any reason to make it `public'?

Comment (sahanHe) [reply persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxGenerator.java]: This was initially changed in order to use this in SQL script generation, but as it was not used in the final code, this was corrected in [50c4917](https://github.com/ballerina-platform/persist-tools/pull/137/commits/50c491782e9456a22a362eafd995ed8372ae16b4)

**Comment by sahanHe** [comment]
This PR will be merged after the update4 release.

File: persist-cli-tests/src/test/resources/test-src/input/tool_test_db_push_25/generated/rainier/generated_client.bal [hunk 3]
```diff
@@ -0,0 +1,468 @@
...
+            tableName: `Employee`,
+            fieldMetadata: {
+                empNo: {columnName: "empNo", 'type: string},
+                firstName: {columnName: "firstName", 'type: string},
+                lastName: {columnName: "lastName", 'type: string},
+                birthDate: {columnName: "birthDate", 'type: time:Date},
+                gender: {columnName: "gender", 'type: string},
+                hireDate: {columnName: "hireDate", 'type: time:Date},
+                'departmentDeptNo: {columnName: "departmentDeptNo", 'type: string},
+                'workspaceWorkspaceId: {columnName: "workspaceWorkspaceId", 'type: string}
```
Comment (daneshk) [inline persist-cli-tests/src/test/resources/test-src/input/tool_test_db_push_25/generated/rainier/generated_client.bal:58]: Do we need generated foreign keys to escape?

File: persist-cli-tests/src/test/resources/test-src/input/tool_test_db_push_25/generated/rainier/generated_types.bal [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+public type 'Employee record {|
+    readonly string empNo;
+    string firstName;
+    string lastName;
+    time:Date birthDate;
+    string gender;
+    time:Date hireDate;
+    string 'departmentDeptNo;
+    string 'workspaceWorkspaceId;
```
Comment (daneshk) [inline persist-cli-tests/src/test/resources/test-src/input/tool_test_db_push_25/generated/rainier/generated_types.bal:46]: Do we need to escape these foreign keys?

**Review by daneshk — APPROVED** [review]
Add a minor comment on foreign keys. Others LGTM

File: persist-cli-tests/src/test/resources/test-src/input/tool_test_db_push_25/generated/rainier/generated_types.bal (hunk 4 above)
Comment (sahanHe) [reply persist-cli-tests/src/test/resources/test-src/input/tool_test_db_push_25/generated/rainier/generated_types.bal:46]: this escape part is coming from the escape in the field type, as we add the field type(entity name) in front to the foreign key it's automatically added to the foreign key

File: persist-cli-tests/src/test/resources/test-src/input/tool_test_db_push_25/generated/rainier/generated_client.bal (hunk 3 above)
Comment (sahanHe) [reply persist-cli-tests/src/test/resources/test-src/input/tool_test_db_push_25/generated/rainier/generated_client.bal:58]: updated in [ab8528a](https://github.com/ballerina-platform/persist-tools/pull/137/commits/ab8528a6d28a13e761569ec8c5a4239d4fb57963)

File: persist-cli-tests/src/test/resources/test-src/input/tool_test_db_push_25/generated/rainier/generated_types.bal (hunk 4 above)
Comment (sahanHe) [reply persist-cli-tests/src/test/resources/test-src/input/tool_test_db_push_25/generated/rainier/generated_types.bal:46]: updated in [ab8528a](https://github.com/ballerina-platform/persist-tools/pull/137/commits/ab8528a6d28a13e761569ec8c5a4239d4fb57963)

**Review by niveathika — DISMISSED** [review]

**Review by niveathika — APPROVED** [review]

**Pull request merged**